tcblib/db_functions.py

```diff
@@ -0,0 +1,97 @@
+import boto3
+import datetime
+import os
+
+DB = boto3.client('dynamodb')
+TABLE = os.environ['DYNAMODB_TABLE']
+DEFAULT_HOST_ID = os.environ['DEFAULT_HOST_ID']
+DEFAULT_HOST_NAME = os.environ['DEFAULT_HOST_NAME']
+
+def get_date():
+    """Get date for next tech chat (will return today if today is Friday)."""
+    today = datetime.date.today()
+    friday = today + datetime.timedelta((4-today.weekday()) % 7)
+    return friday.strftime('%Y%m%d')
```

Comment:
Most people submit gold stars on Friday morning. This will make sure those don't get sent to next week's tech chat. It does mean that stars will be lost after the DM is sent to the host but before midnight but the rate of last-minute stars has gone WAY down. And we can set up like a normal slackbot reminder about that or something, nbd.

Reply:
I think you can solve that. You just need to keep track of the hour, too. `datetime` isnt the easiet to work with, but you can handle it. Its just like, if its past 3pm on friday, punt til next friday.

Reply:
Oh I see, the "date" assigned to the gold star is a "due date": the gold star is read aloud that day.

I think if you make a slight change to your pattern, you'd find this logic was a lot easier. If you insert new gold stars with date set to datetime.now() (i.e. the time it came in at) you could, on gold star report generation (i.e. figuring out what gold stars are for today's tech chat) figure out the bounds of _this_ tech chat (i.e. after 3:00pm previous friday and this upcoming friday) and select goldstars between that range.

Reply:
I think you had it right the first time, actually -- the stars aren't assigned a 'date' so much as the date [which is the primary key i'm using in dynamo] is assigned stars. if i include time, i can add stars to the next week if they come in too late [after the slack message has been sent].

hrm.

Reply:
I think both of my comments are right, one is just more in line with your current thinking. I think it'd be cleaner to record the time at which the stars come in at, and then get stars within a certain range. The datetime can still be the primary key.